src/components/Login/index.jsx

```diff
@@ -1,26 +1,39 @@
 import { useState } from "react";
+import { useNavigate } from "react-router-dom";
 import { collection, addDoc } from "firebase/firestore";
-import { createUserWithEmailAndPassword } from "firebase/auth";
+import { createUserWithEmailAndPassword, signInWithEmailAndPassword } from "firebase/auth";
 
 import { db, auth } from "../../firebase-config";
 
 import "./Login.css"
 
 const Login = () => {
-  const [ data, setData ] = useState(
-    {
-      name: "",
-      phone: "",
-      username: "",
-      email: "",
-      password: ""
-    });
+  const [ data, setData ] = useState({
+    auth_user_id: "",
+    name: "",
+    phone: "",
+    username: "",
+    email: "",
+    password: ""
+  });
+
+  const [ loginData, setLoginData ] = useState({
+    email: "",
+    password: "",
+  });
+
+  const nav = useNavigate();
 
   const changeHandlerData = (event) => {
     const { name, value } = event.target;
     setData({ ...data, [name]: value });
```

Comment:
can we try this?
`{...data, name: value }`

Reply:
No. I don't think so. Because [name] represent dynamic keys within hash.